### ISSUE TITLE
internal/frontend/{cli,cli-ie,qml/BridgeUI}: clarify text of frontendCLI.notifyCertIssue

### ISSUE THREAD

**Issue by kortschak** [ProtonMail/proton-bridge#106]
## Expected Behavior

Provide security warning with adequate information to make informed decision when a TLS certificate error is encountered.

## Current Behavior

A warning that does not clearly state whether there has been information leakage (see [here](https://github.com/ProtonMail/proton-bridge/blob/4912c27be8abb88947cd57743242b8b958c10792/internal/frontend/cli/utils.go#L106-L123) and [here](https://github.com/ProtonMail/proton-bridge/blob/4912c27be8abb88947cd57743242b8b958c10792/internal/frontend/cli-ie/utils.go#L106-L123) — maybe this could be deduplicated? — and [here](https://github.com/ProtonMail/proton-bridge/blob/4912c27be8abb88947cd57743242b8b958c10792/internal/frontend/qml/BridgeUI/DialogTLSCertInfo.qml#L23-L48))

```
Connection security error: Your network connection to Proton services may
be insecure.

Description:
ProtonMail Import-Export was not able to establish a secure connection to Proton
servers due to a TLS certificate error. This means your connection may
potentially be insecure and susceptible to monitoring by third parties.

Recommendation:
* If you trust your network operator, you can continue to use ProtonMail
  as usual.
* If you don't trust your network operator, reconnect to ProtonMail over a VPN
  (such as ProtonVPN) which encrypts your Internet connection, or use
  a different network to access ProtonMail.
```

If the bridge refuses to connect when there is no secure method, then the caveats about trusting network operators are moot. If it doesn't refuse then there are problems since no-one should ever trust their network operator.

## Possible Solution

State in the warning whether the connection has fallen back to an insecure protocol or not.

## Steps to Reproduce

1. Wait for a TLS certificate error...
2.
3.
4.

## Context (Environment)

I expect adequate information to make security decisions.

## Detailed Description

Stated above.

## Possible Implementation

Stated above.

**Comment by JulienPalard** [comment]
Just got this one too:
![proton](https://user-images.githubusercontent.com/239510/96966267-c93e6600-150d-11eb-9c2f-6c081ad45245.png)
I hope protonmail-bridge does *not* use the insecure connection if a secure connection can be established, yet the message make me think so.... :,-(

**Comment by exander77** [comment]
@kortschak 
> If the bridge refuses to connect when there is no secure method, then the caveats about trusting network operators are moot. If it doesn't refuse then there are problems since no-one should ever trust their network operator.

I would note that, that you yourself might be the network operator that is doing SSL termination, for example on your home router. But the description is pretty vague. It should display a certificate signature to either accept or refuse.

**Comment by JulienPalard** [comment]
+1 for the "accept / refuse" button.

With an explicit wording like "Accept to connect insecurly only if you fully understand why this is happening, if any doubt, refuse." e.g. if you're terminating the ssl yourself, you understand why this is popping, and you can click yes.

 (sorry for my english :p)

**Comment by kortschak** [comment]
>I would note that, you yourself might be the network operator that is doing SSL termination, for example on your home router. But the description is pretty vague. It should display a certificate signature to either accept or refuse.

Or allow a prior policy in the settings.

**Comment by JulienPalard** [comment]
So, I have the error again. do Protonmail have certificates issues, or do my network spy on me and Protonmail is gladly allowing it?

This red banner looks like "You're currently spyed by your network, and we're allowing it, you've been warned, use ProtonVPN.".

**Comment by exander77** [comment]
@JulienPalard You may be spying on yourself. What OS do you use? Some Windows Antivirus/Firewalls do SSL termination.

**Comment by exander77** [comment]
@JulienPalard Who, I wanted to write, I never had such error, but I have it as well. Just now.
![Screenshot_2020-10-24_16-56-39](https://user-images.githubusercontent.com/2256039/97084864-0a23a100-161a-11eb-8de6-8762b48123a8.png)

**Comment by exander77** [comment]
This error is really so vague.

**Comment by JulienPalard** [comment]
> @JulienPalard You may be spying on yourself. What OS do you use? Some Windows Antivirus/Firewalls do SSL termination.

I'm on Debian bullseye, with no russian spy^Wantivirus on my machine.

I'm a tad reassured I'm not alone having the issue, it's probably Protonmail side.

**Comment by exander77** [comment]
@JulienPalard Seems that way as I have it exactly the same time as you. I am on Ubuntu 20.10.

**Comment by kortschak** [comment]
I have sent a bug via the bridge about the certificate failures as I have seen more. I think that is a separate though related issue to this; this issue is about the warning behaviour and error message dealing with that problem.

**Comment by kortschak** [comment]
I have seen this behaviour again using v1.4.5, earlier today.

**Comment by jameshoulahan** [comment]
Hi @kortschak, these false positives have been fixed in v1.5.0-beta; you shouldn't see this any more upon updating.

**Comment by exander77** [comment]
@jameshoulahan I have been seen that a lot, but not since I update to v1.5.0 yesterday. I will report if anything suspicious happens.

**Comment by andrzejsza** [comment]
this was finally released fixed with the 2.4.3 beta:
we've changed the test to:
`Bridge cannot verity the authenticity of Proton servers on your current network due to a TLS certificate error. Start Bridge again after ensuring your connection is secure and/or connecting to a VPN. Learn more about TLS pinning here [https://proton.me/blog/tls-ssl-certificate#Extra-security-precautions-taken-by-ProtonMail]`
and the expected behaviour is to not exchange any information with API until the error is removed. no false-positive expected.
